tls/s2n_handshake_io.c

```diff
@@ -651,6 +652,53 @@ static int s2n_conn_update_handshake_hashes(struct s2n_connection *conn, struct
     return 0;
 }
 
+/* this hook runs before hashes are updated */
+static int s2n_conn_pre_handshake_hashes_update(struct s2n_connection *conn)
+{
+    if (conn->actual_protocol_version < S2N_TLS13) return 0;
+
+    switch(s2n_conn_get_current_message_type(conn)) {
```

Comment:
Are we going to add more cases? Otherwise 

    if (s2n_conn_get_current_message_type(conn) != CLIENT_FINISHED) { 
        return 0;
    }

might be simpler.